packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -332,6 +332,74 @@ abstract class ImageProvider<T> {
     return ImageStream();
   }
 
+  /// Checks if this [ImageProvider] has a matching key in the
+  /// [PaintingBinding.imageCache]
+  ///
+  /// The `cache` and `configuration` parameters must not be null. If the
+  /// `handleError` parameter is null, errors will be reported to
+  /// [FlutterError.onError], and the method will return null.
+  ///
+  /// A completed return value of null indicates that an error has occurred.
+  Future<ImageCacheLocation> findCacheLocation({
```

Comment:
It's a bit weird that this returns a Future of ImageCacheLocation, but ImageCacheLocation represents state at a current time.  What time that represents is ambiguous.  A synchronous call or an ImageCacheLocation object that returns the current state is more useful in my estimation.

Reply:
It has to resolve the key, which might involve async work.

Reply:
Image cache location could be defined like this:
```dart
class ImageCacheLocation {
  Object _key;
  final Map<Object, _PendingImage> _pendingImages;
  final Map<Object, _CachedImage> _cache;
  final Map<Object, ImageStreamCompleter> _liveImages;
  bool get pending => _pendingImages.contains(_key);
  // ...
}```

That would make the timing less ambiguous.  So if you ask `pending` it means not right now, not some time between when you ask for it and when you get it.

Reply:
I did some thinking on my walk to the bus.  I guess if it means the position when the future returns it's not so bad.  Feel free to ignore the comments about this.

Reply:
The future in this method is all about getting the key safely.  If you have the key you can get the location synchronously